pyproject.toml

```diff
@@ -1,10 +1,192 @@
+[project]
+name = 'jwst'
+description = 'Library for calibration of science observations from the James Webb Space Telescope'
+readme = 'README.md'
+requires-python = '>=3.8'
+license = { file = 'LICENSE' }
+authors = [{ name = 'JWST calibration pipeline developers' }]
+classifiers = [
+    'Intended Audience :: Science/Research',
+    'Topic :: Scientific/Engineering :: Astronomy',
+    'License :: OSI Approved :: BSD License',
+    'Operating System :: MacOS :: MacOS X',
+    'Operating System :: POSIX',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3 :: Only',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+]
+dependencies = [
+    'asdf >=2.12.0',
+    'asdf-astropy >=0.2.1',
+    'astropy >=5.0.4',
+    'BayesicFitting >=3.0.1',
+    'certifi==2022.5.18.1',
+    'crds >=11.16.5',
+    'drizzle >=1.13.4',
+    'gwcs >=0.18.0',
+    'jsonschema >=4.0.1',
+    'numpy >=1.20',
+    'photutils >=1.4.0',
+    'psutil >=5.7.2',
+    'poppy >=1.0.2',
+    'pyparsing >=2.2.1',
+    'requests >=2.22',
+    'scikit-image >=0.17.2',
+    'scipy >=1.6.0',
+    'spherical-geometry >=1.2.22',
+    'stcal >=1.0',
+    'stdatamodels >=0.4.3, <1.0',
+    'stpipe >=0.4.1, <1.0',
+    'stsci.image >=2.3.5',
+    'stsci.imagestats >=1.6.3',
+    'tweakwcs >=0.7.2',
+    'wiimatch >=0.2.1',
+]
+dynamic = ['version']
+
+[project.optional-dependencies]
+aws = ['stsci-aws-utils >=0.1.2']
+docs = [
+    'packaging',
+    'matplotlib',
+    'mistune ~=0.8.4',
+    'sphinx',
+    'sphinx-asdf >=0.1.1',
+    'sphinx-astropy',
+    'sphinx-automodapi',
+    'sphinx-rtd-theme',
+    'stsci-rtd-theme',
+    'toml',
+]
+sdp = [
+    'jplephem >=2.9',
+    'pymssql',
+    'pysiaf',
+]
+test = [
+    'ci-watson >=0.5.0',
+    'codecov >=1.6.0',
+    'colorama >=0.4.1',
+    'flake8 >=3.6.0',
+    'getch >=1.0.0',
+    'pyproject-flake8',
+    'pytest >=6.0.0',
+    'pytest-cov >=2.9.0',
+    'pytest-doctestplus >=0.10.0',
+    'pytest-openfiles >=0.5.0',
+    'requests_mock >=1.0',
+]
+
+[project.urls]
+'documentation' = 'https://jwst-pipeline.readthedocs.io/en/stable/'
+'repository' = 'https://github.com/spacetelescope/jwst'
+'tracker' = 'https://github.com/spacetelescope/jwst/issues'
+
+[project.entry-points]
+'asdf.extensions' = { jwst_pipeline = 'jwst.transforms.integration:get_extensions' }
+asdf_extensions = { jwst_datamodel = 'jwst.datamodels.extension:DataModelExtension' }
+'asdf.resource_mappings' = { jwst_pipeline = 'jwst.transforms.integration:get_resource_mappings' }
+'stpipe.steps' = { jwst = 'jwst.stpipe.integration:get_steps' }
+pytest11 = { report_crds_context = 'pytest_crds.plugin' }
+
 [build-system]
 requires = [
-    "setuptools>=42",
-    "setuptools_scm[toml]>=3.4",
-    "wheel",
-    "oldest-supported-numpy",
+    'setuptools >=42',
```

Comment:
It looks as if this PR [requires](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html) setuptools 61.0.0 or later to allow pyproject.toml to replace setup.cfg.